src/hooks/usePlants.js

```
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react'
+import getPlants from '../services/getPlants'
+import searchPlants from '../services/searchPlants'
+
+export default function usePlants() {
+    const [plants, setPlants] = useState([])
+    const [searchTerm, setSearchTerm] = useState('')
+    const filteredPlants = filterPlants(searchTerm)
+
+    useEffect(() => {
+        filteredPlants.length === 0 &&
```

for readability I'd write this differently. With the principle of 'fail early' or 'escape early'

```
if (filteredPlants.length !== 0 || !searchTerm.length > 0) {
 return
}
searchPlants(searchTerm)
.blabla()

```